Core/include/Acts/Seeding/Seedfinder.ipp

```diff
@@ -277,20 +277,127 @@ void Seedfinder<external_spacepoint_t, platform_t>::createSeedsForGroup(
+              cosTheta * std::sqrt(1 + A0 * A0)};
+
+          double rMTransf[3];
+          if (!xyzCoordinateCheck(m_config, spM, positionMiddle,
+                                  m_config.toleranceParam, rMTransf)) {
+            continue;
+          }
+
+          // coordinate transformation and checks for bottom spacepoint
+          float B0 = 2. * (Vb - A0 * Ub);
+          float Cb = 1. - B0 * lb.y;
+          float Sb = A0 + B0 * lb.x;
+          double positionBottom[3] = {
+              rotationTermsUVtoXY[0] * Cb - rotationTermsUVtoXY[1] * Sb,
+              rotationTermsUVtoXY[0] * Sb + rotationTermsUVtoXY[1] * Cb,
+              cosTheta * std::sqrt(1 + A0 * A0)};
+
+          auto spB = state.compatBottomSP[b];
+          double rBTransf[3];
+          if (!xyzCoordinateCheck(m_config, spB, positionBottom,
+                                  m_config.toleranceParam, rBTransf)) {
+            continue;
+          }
+
+          // coordinate transformation and checks for top spacepoint
+          float Ct = 1. - B0 * lt.y;
+          float St = A0 + B0 * lt.x;
+          double positionTop[3] = {
+              rotationTermsUVtoXY[0] * Ct - rotationTermsUVtoXY[1] * St,
+              rotationTermsUVtoXY[0] * St + rotationTermsUVtoXY[1] * Ct,
+              cosTheta * std::sqrt(1 + A0 * A0)};
+
+          auto spT = state.compatTopSP[t];
+          double rTTransf[3];
+          if (!xyzCoordinateCheck(m_config, spT, positionTop,
+                                  m_config.toleranceParam, rTTransf)) {
+            continue;
+          }
+
+          // correcting other variables
+          float xB = rBTransf[0] - rTTransf[0];
+          float yB = rBTransf[1] - rTTransf[1];
+          float zB = rBTransf[2] - rTTransf[2];
+          float xT = rBTransf[0] - rTTransf[0];
+          float yT = rBTransf[1] - rTTransf[1];
+          float zT = rBTransf[2] - rTTransf[2];
+
+          float iDeltaRB2 = 1. / (xB * xB + yB * yB);
+          float iDeltaRT2 = 1. / (xT * xT + yT * yT);
+
+          cotThetaBNew = -zB * std::sqrt(iDeltaRB2);
+          cotThetaT = zT * std::sqrt(iDeltaRT2);
+
+          rMxy =
+              std::sqrt(rMTransf[0] * rMTransf[0] + rMTransf[1] * rMTransf[1]);
+          float Ax = rMTransf[0] / rMxy;
+          float Ay = rMTransf[1] / rMxy;
+
+          ub = (xB * Ax + yB * Ay) * iDeltaRB2;
+          vb = (yB * Ax - xB * Ay) * iDeltaRB2;
+          ut = (xT * Ax + yT * Ay) * iDeltaRT2;
+          vt = (yT * Ax - xT * Ay) * iDeltaRT2;
+
+        } else {
+          // don't use the xyz coordinate check
+          cotThetaBNew = cotThetaB;
+        }
+
+        float cotThetaAvg2;
+        if (m_config.arithmeticAverageCotTheta) {
+          // use arithmetic average
+          cotThetaAvg2 = std::pow((cotThetaBNew + lt.cotTheta) / 2, 2);
+        } else {
+          // use geometric average
+          cotThetaAvg2 = cotThetaBNew * lt.cotTheta;
```

Comment:
```suggestion
          cotThetaAvg2 = cotThetaBNew * cotThetaT;
```
to make it clear it's the same as below :)

Reply:
can you replace it everywhere?

Reply:
Agree, that should be cotThetaT

Reply:
Done!